### PR TITLE
feat: Feeds Section | Enable Click Functionality for Trending Tags and Comments in Posts

### PR DESCRIPTION
**Pull request by Som401**
## Fixes Issue

related to issue #626

## Changes proposed

I have made the following enhancements to address the issues in the Feeds section:

### 1) Comment Icon in Posts
- Clicking on the comment icon in each post now opens the comment section of that specific post.

### 2) Trending Tags Functionality
- Implemented functionality for trending tags in the sidebar.
  - Clicking on a tag filters the posts based on the chosen tag.
  - Clicking on another tag removes the previous tag and filters the posts based on the new one.
  - If the same tag is clicked again, it is removed from the filter.

#### Note:
I attempted to implement a function to filter multiple tags, but encountered an issue when tag names contain spaces. This functionality is currently not supported due to this limitation.

## Code of Conduct

- [x] By submitting this pull request, I confirm I've read and complied with the [CoC](https://github.com/thecyberworld/Support/blob/main/CODE_OF_CONDUCT.md) 🖖

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

---

You can also join our [Discord](https://discord.gg/QHBPq6xP5p) community.
Feel free to check out other cool repositories of the [Thecyberworld](https://github.com/thecyberworld).
Join the Thecyberworld GitHub Organisation by raising an [issue](https://github.com/thecyberworld/Support/issues/new?assignees=&labels=invite+me+to+the+organisation&template=invitation.yml&title=Please+invite+me+to+the+GitHub+Community+Organization) (you will be sent an invitation).

### CODE REVIEW

**Comment by ArkadiK94** [comment]
@Som401 could you add screen shot/ short video?

**Comment by Som401** [comment]
https://github.com/thecyberworld/TheCyberHUB/assets/147496504/73ec0af0-4069-4892-b165-c7dad87c2e85

@ArkadiK94 sorry for being late

File: src/components/Feeds/FeedTags/FeedTagsElements.jsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ export const Tag = styled.p`
     word-break: break-word;
     overflow-wrap: break-word;
     text-transform: capitalize;
+    user-select: none;
```
Comment (ArkadiK94) [inline src/components/Feeds/FeedTags/FeedTagsElements.jsx:48]: Wow, I learned a new css property that I didn't know that was exist haha.

**Review by ArkadiK94 — APPROVED** [review]
@Som401 Great work, thanks for your contribution.

**Comment by ArkadiK94** [comment]
> I attempted to implement a function to filter multiple tags, but encountered an issue when tag names contain spaces. This functionality is currently not supported due to this limitation.

Hmm, what about using trim() to remove the spaces? or did you mean something else?

File: src/components/Feeds/FeedTags/FeedTagsElements.jsx (hunk 1 above)
Comment (Som401) [reply src/components/Feeds/FeedTags/FeedTagsElements.jsx:48]: It's great that you discovered a new property. By the way, I've noticed that many components in the project could benefit from this property too.

Comment (ArkadiK94) [reply src/components/Feeds/FeedTags/FeedTagsElements.jsx:48]: @kabir0x23

**Comment by ArkadiK94** [comment]
>I attempted to implement a function to filter multiple tags, but encountered an issue when tag names contain spaces. This functionality is currently not supported due to this limitation.


@Som401 I will merge this pr and you can open a new one for implementation of filter multiple tags.

**Comment by Som401** [comment]
> > I attempted to implement a function to filter multiple tags, but encountered an issue when tag names contain spaces. This functionality is currently not supported due to this limitation.
> 
> Hmm, what about using trim() to remove the spaces? or did you mean something else?

Ah, I see. Thanks for clarifying. When I mentioned facing issues with tag names containing spaces, I wasn't referring to the conventional leading or trailing spaces that can be handled with the trim() function. Instead, the challenge is about dealing with multi-word tag names such as "security logo." 

To tackle this, I've taken a screenshot of the function to better illustrate the problem.

capture

After pushing such a tag ("security logo") to the searchTerm string and attempting to remove it by clicking on the tag, the function splits the searchTerm into an array using spaces as separators. For example, "security logo" becomes ["security", "logo"]. However, this array doesn't match the original tag name "security logo," causing the function to mistakenly add the tag back each time it's clicked. I even tried with splitting the tag name and comparing substrings, but it introduced new issues, especially with single-word tags like "security."

Considering this, I'm proposing an alternative approach: enforcing a naming convention that replaces spaces with underscores. ("security_logo" instead of "security logo") 

Let me know your thoughts on this approach or if you have any other suggestions or perhaps a better algorithm to solve this issue!

**Pull request merged**